Framework/API/src/WorkspaceFactory.cpp

```diff
@@ -51,16 +51,23 @@ MatrixWorkspace_sptr
 WorkspaceFactoryImpl::create(const MatrixWorkspace_const_sptr &parent,
                              size_t NVectors, size_t XLength,
                              size_t YLength) const {
-  bool differentSize(true);
+  bool differentSize(false);
```

Comment:
Now the flag can be `false` only if the numbers of histogram did not change, which will change other observable behavior of `create`. In particular bin masking will not be copied any more (see `initializeFromParent` below) if the number of histograms changed. I am not saying this is the wrong thing to do but I cannot really make a call. Maybe this also warrants a wider discussion?

Reply:
From what I can tell bin masking is the only other time `differentSize` is used. That being the case it makes sense to me to remove this parameter and use `parent->blocksize() != child->blocksize()` as the test for copying bin masking.

For axes the use of `differentSize` makes very little sense to me as if the X axis does not change then you end up copying all axes from the parent, which leads to invalid child workspaces.

Reply:
The issue around the bin masks is that the map being copied contains a lookup against the workspace index so if you are changing the number of histograms the map is really not valid either. I have to say I think I would be very nervous at changing this behaviour.